src/main/java/com/extendedclip/deluxemenus/menu/Menu.java

```diff
@@ -488,6 +495,11 @@ public void openMenu(final Player viewer, final Map<String, String> args, final
           holder.startUpdatePlaceholdersTask();
         }
       });
+
+      Bukkit.getScheduler().runTask(DeluxeMenus.getInstance(), () -> {

```

Comment:
What's the purpose in doing this?

Reply:
So the event runs on the main thread, I think that is the preferable option here. Can be ran async though, see no problem with that. But the event name would have to include Async or note it in any other way.